### ISSUE TITLE
Wrong number of parameters for nn4.small2.v1

### ISSUE THREAD

**Issue by seonho** [cmusatyalab/openface#125]
### Context of the issue.

The following page shows summary of model parameters for each model.

http://cmusatyalab.github.io/openface/models-and-accuracies/

But, I think the number of parameters for `nn4.small2.v1` model should be `3743280`.
(currently `3733968`)

I also checked pre-trained model for `nn4.small2.v1.t7` and I got a total number of parameter as `3743280`.

### Steps to reproduce.

The size of tensors for layers (convolution, batch normalization, linear) are as follows.

```
9408	
64	
64	
64	
64	
64	
4096	
64	
64
64	
64	
64	
110592	
192	
192	
192	
192	
192	
18432	
96	
96	
96	
96	
96	
110592	
128	
128	
128	
128	
128	
3072	
16	
16	
16	
16	
16	
12800	
32	
32	
32	
32	
32	
6144	
32	
32	
32	
32	
32	
12288	
64	
64	
64	
64	
64	
24576	
96	
96	
96	
96	
96	
110592	
128	
128	
128	
128	
128	
8192	
32	
32	
32	
32	
32	
51200	
64	
64	
64	
64	
64	
16384	
64	
64	
64	
64	
64	
16384	
64	
64	
64	
64	
64	
40960	
128	
128	
128	
128	
128	
294912	
256	
256	
256	
256	
256	
10240	
32	
32	
32	
32	
32	
51200	
64	
64	
64	
64	
64	
61440	
96	
96	
96	
96	
96	
165888	
192	
192	
192	
192	
192	
20480	
32	
32	
32	
32	
32	
51200	
64	
64	
64	
64	
64	
81920	
128	
128	
128	
128	
128	
163840	
256	
256	
256	
256	
256	
102400	
160	
160	
160	
160	
160	
368640	
256	
256	
256	
256	
256	
40960	
64	
64	
64	
64	
64	
204800	
128	
128	
128	
128	
128	
98304	
96	
96	
96	
96	
96	
331776	
384	
384	
384	
384	
384	
98304	
96	
96	
96	
96	
96	
262144	
256	
256	
256	
256	
256	
70656	
96	
96	
96	
96	
96	
331776	
384	
384	
384	
384	
384	
70656	
96	
96	
96	
96	
96	
188416	
256	
256	
256	
256	
256	
94208	
128
```

**Comment by bamos** [comment]
Hi @seonho - thanks for reporting this. Can you post the code you used to get those numbers? Both `torch.getParameters` and `torch.parameters` are returning the values I posted on the website, but I assume it's not including some parameters with different names. 

```
th> net = torch.load('nn4.small2.v1.t7')
th> params, gradParams = net:getParameters()
th> params:size()
 3733968
[torch.LongStorage of size 1]
```

```
th> params, gParams = net:parameters()
th> sum = 0
th> for i, p in ipairs(params) do
  sum = sum + p:view(-1):size(1)
end
th> sum
3733968
```

-Brandon.

**Comment by seonho** [comment]
I considered three types of layer: convolution, batch normalization, and linear.
And four tensors - weight, bias, running_mean, running_var - are investigated.
Some constant parameters are ignored.

Here is my test code.

```lua
require 'nn'
require 'dpnn'

function print_weight_and_bias(layer)
    print(layer.weight:nElement())
    print(layer.bias:nElement())
end

local M = {}

M.CONVERTER = {
    ['nn.SpatialConvolutionMM'] = function(container, layer)
        print_weight_and_bias(layer)
    end,
    ['nn.SpatialConvolution'] = function(container, layer)
        print_weight_and_bias(layer)
    end,
    ['nn.SpatialBatchNormalization'] = function(container, layer)
        print_weight_and_bias(layer)
        print(layer.running_mean:nElement())
        print(layer.running_var:nElement())
    end,
    ['nn.SpatialConvolutionMM'] = function(container, layer)
        print(layer.weight:nElement())
        print(layer.bias:nElement())
    end,
    ['nn.Sequential'] = function(container, layer)
        subcont = {}
        for i = 1, #layer.modules do
            M.add(subcont, layer.modules[i])
        end
    end,
    ['nn.Inception'] = function(container, layer)
        subcont = {}
        for i = 1, #layer.modules do
            M.add(subcont, layer.modules[i])
        end
    end,
    ['nn.DepthConcat'] = function(container, layer)
        subcont = {}
        for i = 1, #layer.modules do
            M.add(subcont, layer.modules[i])
        end
    end,
    ['nn.Linear'] = function(container, layer)
        print_weight_and_bias(layer)
    end,
}

function M.add(net, layer)
    local layer_type = torch.type(layer)
    for layer_pattern, converter in pairs(M.CONVERTER) do
        if string.find(layer_type, layer_pattern) then
            return converter(net, layer)
        end
    end
end

-- load model
model = torch.load('nn4.small2.v1.t7')
for i = 1, #model.modules do
    M.add(net, model.modules[i])
end
```

**Comment by bamos** [comment]
Thanks for the code @seonho! The running mean and variance in the batch normalization is causing the difference between your code and `getParameters`. After removing these I get the same parameter count as before. With your code in `count.lua`:

```
openface(master*)$ awk '{s+=$1} END {print s}' <(./count.lua)
3733968
```

I've added a clarifying note to the website that the parameter counts don't include the batch normalization means and variances.

-Brandon.
